Userland/Libraries/LibC/unistd.cpp

```diff
@@ -891,6 +902,8 @@ int sysbeep()
 // https://pubs.opengroup.org/onlinepubs/9699919799/functions/fsync.html
 int fsync(int fd)
 {
+    __pthread_maybe_cancel();
```

Comment:
Putting this in fsync seems like a ticking time bomb of a bug? 😀

Reply:
I was about to say "POSIX says this is a thing", but the selection of cancellation points I actually based on the [Linux manpage](https://man7.org/linux/man-pages/man7/pthreads.7.html). POSIX doesn't seem to have an equivalent to this.

Yes, this could potentially delay flushing the file, at least until some other thread gets to it or until the process exits as a whole.